zebrad/src/components/inbound.rs

```diff
@@ -145,64 +152,58 @@ impl Service<zn::Request> for Inbound {
         // and reporting unreadiness might cause unwanted load-shedding, since
         // the load-shed middleware is unable to distinguish being unready due
         // to load from being unready while waiting on setup.
-        if matches!(self.network, Setup::AwaitingNetwork { .. }) {
-            // Unfortunately, we can't match, swap, and destructure at the same time
-            let mut awaiting_state = Setup::FailedInit;
-            mem::swap(&mut self.network, &mut awaiting_state);
-            if let Setup::AwaitingNetwork {
+        let mut result = Ok(());
+
+        self.network = match self.take_setup() {
```

Comment:
Would it be better to assign to `(self.network, result)` here?

That way, each match arm has an explicit result, rather than defaulting to success, and accidentally missing an error case.

Reply:
> error[E0658]: destructuring assignments are unstable

😢

Reply:
Fixed in 4f172ab6747de137fc8c1ead27290b4c27816ad5